File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/configuration/ChatCommandConfiguration.java
package net.bestia.zoneserver.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import net.entity.factory.MobFactory;
import bestia.messages.MessageApi;
import net.bestia.zoneserver.chat.ChatCommandService;
import net.bestia.zoneserver.chat.MetaChatCommand;
import net.bestia.zoneserver.chat.MobSpawnModule;

/**
 * Assembles the meta chat commands so they are getting picked up by the
 * {@link ChatCommandService}.
 * 
 * @author Thomas Felix
 *
 */
@Configuration
public class ChatCommandConfiguration {

	@Bean
	public MetaChatCommand getSetChatCommand() {

		final MetaChatCommand setCmd = new MetaChatCommand("/entity");

		return setCmd;
	}

	@Bean
	public MetaChatCommand getSpawnChatCommand(MessageApi akkaApi, MobFactory mobFactory) {

		final MetaChatCommand spawnCmd = new MetaChatCommand("/spawn");

		final MobSpawnModule mobModule = new MobSpawnModule(akkaApi, mobFactory);

		spawnCmd.addCommandModule(mobModule);

		return spawnCmd;
	}

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/configuration/ChatCommandConfiguration.kt
package net.bestia.zoneserver.configuration

import net.bestia.entity.factory.MobFactory
import net.bestia.zoneserver.chat.ChatCommandService
import net.bestia.zoneserver.chat.MetaChatCommand
import net.bestia.zoneserver.chat.MobSpawnModule
import org.springframework.context.annotation.Configuration

/**
 * Assembles the meta chat commands so they are getting picked up by the
 * [ChatCommandService].
 *
 * @author Thomas Felix
 */
@Configuration
class ChatCommandConfiguration {

  // @Bean This currently gives a strange exception
  fun getSpawnChatCommand(akkaApi: MessageApi, mobFactory: MobFactory): MetaChatCommand {
    val spawnCmd = MetaChatCommand("/spawn")
    val mobModule = MobSpawnModule(akkaApi, mobFactory)
    spawnCmd.addCommandModule(mobModule)

    return spawnCmd
  }
}
